pkg/apis/apps/validation/validation_test.go

```diff
@@ -62,53 +62,104 @@ func makeSet(selectorLabels *map[string]string, template *api.PodTemplate, tweak
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Labels = labels
 	}
 }
 
-func tweakRepicas(replicas int32) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.Replicas = replicas
+func tweakAnnotations(annotations map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Annotations = annotations
 	}
 }
 
-func tweakPodManagementPolicy(policy apps.PodManagementPolicyType) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.PodManagementPolicy = policy
+func tweakFinalizers(finalizers []string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.ObjectMeta.Finalizers = finalizers
 	}
 }
 
-func tweakUpdateStrategy(strategy apps.StatefulSetUpdateStrategy) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.UpdateStrategy = strategy
+func tweakManagementPolicy(policy apps.PodManagementPolicyType) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.PodManagementPolicy = policy
 	}
 }
 
-func tweakMinReadySeconds(t int32) setTweak {
-	return func(s apps.StatefulSet) {
-		s.Spec.MinReadySeconds = t
+func tweakReplicas(replicas int32) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.Replicas = replicas
+	}
+}
+
+func tweakPodManagementPolicy(policy apps.PodManagementPolicyType) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.PodManagementPolicy = policy
+	}
+}
+
+func tweakSelectorLabels(labels map[string]string) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.Selector = &metav1.LabelSelector{MatchLabels: labels}
+	}
+}
+
+func tweakRestartPolicy(rp api.RestartPolicy) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.Template.Spec.RestartPolicy = rp
+	}
+}
+
+func tweakMinReadySeconds(t int32) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.MinReadySeconds = t
+	}
+}
+
+func tweakOrdinalsStart(s int32) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.Ordinals = &apps.StatefulSetOrdinals{Start: s}
+	}
+}
+
+func tweakPVCPolicy(policy apps.StatefulSetPersistentVolumeClaimRetentionPolicy) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.PersistentVolumeClaimRetentionPolicy = &policy
+	}
+}
+
+func tweakUpdateStrategy(strategy apps.StatefulSetUpdateStrategy) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.UpdateStrategy = strategy
+	}
+}
+
+func tweakPVCTemplateSize(pvc []api.PersistentVolumeClaim) statefulSetTweak {
+	return func(ss apps.StatefulSet) {
+		ss.Spec.VolumeClaimTemplates = pvc
 	}
 }
 
 type updateStrategyTweak func(s apps.StatefulSetUpdateStrategy)
 
-func makeUpdateStrategy(tweaks ...updateStrategyTweak) apps.StatefulSetUpdateStrategy {
+func mkUpdateStrategy(tweaks ...updateStrategyTweak) apps.StatefulSetUpdateStrategy {
 	s := apps.StatefulSetUpdateStrategy{
 		Type: apps.RollingUpdateStatefulSetStrategyType,
+		RollingUpdate: func() *apps.RollingUpdateStatefulSetStrategy {
```

Comment:
Why this `func` construct?  I'm missing the purpose of it.

Reply:
removed